File: php/tests/PrimeFactors/PrimeFactorsTest.php
<?php
namespace pdt256\kata\PrimeFactors;

class PrimeFactorsTest extends \PHPUnit_Framework_TestCase
{
    /** @var PrimeFactorGeneratorInterface */
    private $primeFactors;

    public function setUp()
    {
        set_time_limit(1);
        $this->primeFactors = new PrimeFactors;
    }

    public function testGenerate()
    {
        $this->assertSame([], $this->primeFactors->generate(1));
        //$this->assertSame([2], $this->primeFactors->generate(2));
    }

    /**
     * @param int $number
     * @param array $primes
     * @dataProvider primeFactorsData
     */
    public function xtestFinal(int $number, array $primes)
    {
        $result = $this->primeFactors->generate($number);
        $this->assertEquals(
            $primes,
            $result,
            'Input: ' . $number . ', Expected: ' . json_encode($primes) . ', Received: ' . json_encode($result)
        );
    }

    public function primeFactorsData()
    {
        return [
            [1, []],
            [2, [2]],
            [3, [3]],
            [4, [2, 2]],
            [5, [5]],
            [6, [2, 3]],
            [7, [7]],
            [8, [2, 2, 2]],
            [9, [3, 3]],
            [360, [2, 2, 2, 3, 3, 5]],
            [(2 * 3 * 5 * 7 * 13), [2, 3, 5, 7, 13]],
        ];
    }
}

File: go/pkg/primefactors/primefactors_test.go
package primefactors_test

import (
	"testing"

	"github.com/stretchr/testify/assert"

	"github.com/pdt256/kata/go/pkg/primefactors"
)

func TestPrimeFactors(t *testing.T) {
	assert.Equal(t, []int(nil), primefactors.Generate(1))
	// assert.Equal(t, []int{2}, primefactors.Generate(2))
}

func TestAllPrimeFactors(t *testing.T) {
	t.Skip()
	for _, data := range primeFactorsData {
		assert.Equal(t, data.expected, primefactors.Generate(data.number))
	}
}

var primeFactorsData = []struct {
	number   int
	expected []int
}{
	{1, []int(nil)},
	{2, []int{2}},
	{3, []int{3}},
	{4, []int{2, 2}},
	{5, []int{5}},
	{6, []int{2, 3}},
	{7, []int{7}},
	{8, []int{2, 2, 2}},
	{9, []int{3, 3}},
	{360, []int{2, 2, 2, 3, 3, 5}},
	{2 * 3 * 5 * 7 * 13, []int{2, 3, 5, 7, 13}},
}
